nexus/types/src/external_api/params.rs

```diff
@@ -35,6 +36,31 @@ pub struct InstancePath {
     pub instance: NameOrId,
 }
 
+#[derive(Deserialize, JsonSchema)]
+pub struct NetworkInterfacePath {
+    pub interface: NameOrId,
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct VpcPath {
+    pub vpc: NameOrId,
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct SubnetPath {
+    pub subnet: NameOrId,
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct RouterPath {
+    pub router: NameOrId,
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct RoutePath {
+    pub route: NameOrId,
+}
+
```

Comment:
Not something to be addressed in this PR, but it's a little silly that these are all the same. If we called it `resource` in all cases maybe they could share the struct. Something like:

```rs
#[derive(Deserialize, JsonSchema)]
pub struct NameOrIdPath {
    pub resource: NameOrId,
}
```

Reply:
Yeah, the ultimate intent here I think is to have it be generated by a macro. Unfortunately the name matters as it shows up in documentation, the API spec, etc.

Reply:
We did that at "previous job" (have a common "resource" struct), and it was such a pain when we needed to add additional fields for a new version, it ended up being a spaghetti nightmare 😅